src/sentry/charts/metric_alerts.py

```diff
@@ -0,0 +1,226 @@
+SESSION_AGGREGATE_TO_FIELD = {
+    CRASH_FREE_SESSIONS: "sum(session)",
+    CRASH_FREE_USERS: "count_unique(user)",
+}
+
+
+API_INTERVAL_POINTS_LIMIT = 10000
+API_INTERVAL_POINTS_MIN = 150
+
+
+def incident_date_range(alert_rule: AlertRule, incident: Incident):
+    """Retrieve the start/end for graphing an incident."""
+    time_window_seconds = alert_rule.snuba_query.time_window
+    min_range = time_window_seconds * API_INTERVAL_POINTS_MIN
+    max_range = time_window_seconds * API_INTERVAL_POINTS_LIMIT
+    now = timezone.now()
+    start_date: datetime = incident.date_started
+    end_date: datetime = incident.date_closed if incident.date_closed else now
+    incident_range = max((end_date - start_date).total_seconds(), 3 * time_window_seconds)
+    range = min(max_range, max(min_range, incident_range))
+    half_range = timedelta(seconds=range / 2)
+
+    return {
+        "start": (start_date - half_range).strftime("%Y-%m-%dT%H:%M:%S"),
+        "end": (end_date + half_range).strftime("%Y-%m-%dT%H:%M:%S"),
+    }
+
+
+def fetch_metric_alert_sessions_data(
+    organization: Organization,
+    alert_rule: AlertRule,
+    time_period: Dict[str, Any],
+    user: Optional["User"] = None,
+):
+    aggregate = translate_aggregate_field(alert_rule.snuba_query.aggregate, reverse=True)
+    project_id = alert_rule.snuba_query.subscriptions.select_related("project").first().project.id
+
+    if "period" in time_period:
+        time_period = {"statsPeriod": time_period["period"]}
+
+    env = alert_rule.snuba_query.environment
+    env_params = {"environment": env} if env else {}
+
+    try:
+        resp = client.get(
+            auth=ApiKey(organization=organization, scope_list=["org:read"]),
+            user=user,
+            path=f"/organizations/{organization.slug}/sessions/",
+            params={
+                "field": SESSION_AGGREGATE_TO_FIELD[aggregate],
+                "interval": f"{alert_rule.snuba_query.time_window}m",
+                "project": str(project_id),
+                "query": alert_rule.snuba_query.query,
+                "groupBy": "session.status",
+                **env_params,
+                **time_period,
+            },
+        )
+        return resp.data
+    except Exception as exc:
+        logger.error(
+            f"Failed to load sessions for chart: {exc}",
+            exc_info=True,
+        )
+        return None
+
+
+def fetch_metric_alert_events_timeseries(
+    organization: Organization,
+    alert_rule: AlertRule,
+    time_period: Dict[str, Any],
+    user: Optional["User"] = None,
+):
+    env = alert_rule.snuba_query.environment
+    env_params = {"environment": env} if env else {}
+    aggregate = translate_aggregate_field(alert_rule.snuba_query.aggregate, reverse=True)
+    project_id = alert_rule.snuba_query.subscriptions.select_related("project").first().project.id
+    try:
+        resp = client.get(
+            auth=ApiKey(organization=organization, scope_list=["org:read"]),
```

Comment:
@NisanthanNanthakumar This was based on the existing discover unfurl. I assume we need an API key for the org to access their data? Not totally sure. https://github.com/getsentry/sentry/blob/6766ffc7a2ceec0fafe0aebd378427323ae0138a/src/sentry/integrations/slack/unfurl/discover.py#L115-L118

Reply:
@NisanthanNanthakumar why? What's wrong with using our internal API client?

Reply:
We typically prefer to not use the internal API client to hit other APIs when we could just refactor and share code that way.

In the case of chartcuterie, we actually want to get the exact same api response that the frontend gets so that sharing code is easier.